linode/domain/framework_resource.go

```diff
@@ -0,0 +1,235 @@
+	req resource.ReadRequest,
+	resp *resource.ReadResponse,
+) {
+	client := r.Meta.Client
+
+	var data DomainModel
+
+	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	id := data.ID.ValueInt64()
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	domain, err := client.GetDomain(ctx, int(id))
+	if err != nil {
+		if lerr, ok := err.(*linodego.Error); ok && lerr.Code == 404 {
+			resp.Diagnostics.AddWarning(
+				"Domain No Longer Exists",
+				fmt.Sprintf(
+					"Removing Domain with ID %v from state because it no longer exists",
+					data.ID,
+				),
+			)
+			resp.State.RemoveResource(ctx)
+			return
+		}
+		resp.Diagnostics.AddError(
+			"Error finding the specified Domain",
+			err.Error(),
+		)
+		return
+	}
+
+	data.parseComputed(ctx, domain)
+	data.parseNonComputed(domain)
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (r *Resource) Create(
+	ctx context.Context,
+	req resource.CreateRequest,
+	resp *resource.CreateResponse,
+) {
+	client := r.Meta.Client
+	var data DomainModel
+
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	createOpts := linodego.DomainCreateOptions{
+		Domain:      data.Domain.ValueString(),
+		Type:        linodego.DomainType(data.Type.ValueString()),
+		Group:       data.Group.ValueString(),
+		Description: data.Description.ValueString(),
+		SOAEmail:    data.SOAEmail.ValueString(),
+		RetrySec:    int(data.RetrySec.ValueInt64()),
+		ExpireSec:   int(data.ExpireSec.ValueInt64()),
+		RefreshSec:  int(data.RefreshSec.ValueInt64()),
+		TTLSec:      int(data.TTLSec.ValueInt64()),
+	}
+
+	if !data.MasterIPs.IsNull() && !data.MasterIPs.IsUnknown() {
+		resp.Diagnostics.Append(data.MasterIPs.ElementsAs(ctx, &createOpts.MasterIPs, false)...)
+		if resp.Diagnostics.HasError() {
+			return
+		}
+	}
+	if !data.AXFRIPs.IsNull() && !data.AXFRIPs.IsUnknown() {
+		resp.Diagnostics.Append(data.AXFRIPs.ElementsAs(ctx, &createOpts.AXfrIPs, false)...)
+		if resp.Diagnostics.HasError() {
+			return
+		}
+	}
+	if !data.Tags.IsNull() && !data.Tags.IsUnknown() {
+		resp.Diagnostics.Append(data.Tags.ElementsAs(ctx, &createOpts.Tags, false)...)
+		if resp.Diagnostics.HasError() {
+			return
+		}
+	}
+
+	domain, err := client.CreateDomain(ctx, createOpts)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Domain creation error",
+			err.Error(),
+		)
+		return
+	}
+
+	data.parseComputed(ctx, domain)
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (r *Resource) Update(
+	ctx context.Context,
+	req resource.UpdateRequest,
+	resp *resource.UpdateResponse,
+) {
+	var plan, state DomainModel
+
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	domainID := int(state.ID.ValueInt64())
+
+	if !domainDeepEqual(plan, state) {
+		r.updateDomain(ctx, resp, plan, domainID)
+	}
```

Comment:
```suggestion
	if !domainDeepEqual(plan, state) {
		r.updateDomain(ctx, resp, plan, domainID)
	} else {
		resp.Diagnostics.Append(resp.State.Set(ctx, &plan)...)
	}
```

> An error is returned unless every null or known value in the request plan is saved exactly as-is into the response state.
https://developer.hashicorp.com/terraform/plugin/framework/resources/update#caveats

Although extra effort may be needed to handle Unknown values in the `else` branch, I think it's nicer do this to ensure we align with the requirement from the TF framework's docs.